DigitalLearningSolutions.Data/Helpers/ExternalApis/LearningHubApiClient.cs

```diff
@@ -0,0 +1,142 @@
+﻿namespace DigitalLearningSolutions.Data.Helpers.ExternalApis
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Net.Http;
+    using System.Threading.Tasks;
+    using DigitalLearningSolutions.Data.Models.LearningHubApiClient;
+    using Microsoft.Extensions.Configuration;
+    using Newtonsoft.Json;
+
+    public interface ILearningHubApiClient
+    {
+        Task<ResourceSearchResult> SearchResource(
+            string text,
+            int? offset = null,
+            int? limit = null,
+            IEnumerable<string>? resourceTypes = null
+        );
+
+        Task<ResourceReferenceWithReferenceDetails> GetResourceByReferenceId(int resourceReferenceId);
+
+        Task<BulkResourceReferences> GetBulkResourcesByReferenceIds(
+            IEnumerable<int>? resourceReferenceIds = null
+        );
+    }
+
+    public class LearningHubApiClient : ILearningHubApiClient
+    {
+        private readonly HttpClient client;
+
+        public LearningHubApiClient(HttpClient httpClient, IConfiguration config)
+        {
+            string learningHubApiKey = config.GetLearningHubApiKey();
+            string learningHubApiBaseUrl = config.GetLearningHubApiBaseUrl();
+
+            client = httpClient;
+            client.BaseAddress = new Uri(learningHubApiBaseUrl);
+            client.DefaultRequestHeaders.Accept.Clear();
+            client.DefaultRequestHeaders.Add("X-API-KEY", learningHubApiKey);
+        }
+
+        public async Task<ResourceSearchResult> SearchResource(
+            string text,
+            int? offset = null,
+            int? limit = null,
+            IEnumerable<string>? resourceTypes = null
+        )
+        {
+            var queryParams = GetSearchQueryParams(text, offset, limit, resourceTypes);
+
+            var response = await client.GetStringAsync($"/Resource/Search?{queryParams}");
+            var result = JsonConvert.DeserializeObject<ResourceSearchResult>(response);
+            return result;
+        }
+
+        public async Task<ResourceReferenceWithReferenceDetails> GetResourceByReferenceId(int resourceReferenceId)
+        {
+            var response = await client.GetStringAsync($"/Resource/{resourceReferenceId}");
+            var result = JsonConvert.DeserializeObject<ResourceReferenceWithReferenceDetails>(response);
+            return result;
+        }
+
+        public async Task<BulkResourceReferences> GetBulkResourcesByReferenceIds(
+            IEnumerable<int>? resourceReferenceIds = null
+        )
+        {
+            var queryParams = GetBulkResourcesQueryParams(resourceReferenceIds);
+
+            var response = await client.GetStringAsync($"/Resource/Bulk{queryParams}");

```

Comment:
Not super keen on the fact we've got a different setup here to the Search endpoint. I think the ? should probably be included in this string.
Also this doesn't seem to return anything if we don't enter any parameters, is that the intended behaviour?

Reply:
I didn't include the ? here because it is possible to make the call with an empty array of referenceIds, in which case there is no ? appended onto the url. If you don't enter parameters it should return something like this:
`{
  "resourceReferences": [],
  "unmatchedResourceReferenceIds": []
}`
Is that what you're getting?

Reply:
I'll do a bit more refactoring to make it smarter, and we can have a chat or pair on it tomorrow :)

Reply:
Yeah, but I get the same result if I move the ? here and just don't have anything after it because the private method returns string.Empty.

Reply:
I'm only commenting on the behaviour because that seems strange. Why have a nullable parameter on the API if it returns nothing useful without it.

Reply:
Unless...are we assuming that we'll always call this Bulk method will at least one id? Then I could use the same set up as the search endpoint. I was just going off what [swagger ](https://uks-learninghubnhsuk-openapi-dev.azurewebsites.net/swagger/index.html) allows you to enter, but we'll never call this without parameters right?

Reply:
Since it's an IEnumerable, yeah, we should discourage people from calling this method with `null`. They can always call it with an empty list, which is preferable as a "nothing" case.